### PR TITLE
Add support for Insurance objects

### PR DESCRIPTION
**Pull request by nwithan8**
Finally, our C# library supports Insurance!

This PR includes:
- A new `Insurance` class, with `Create()`, `Retrieve()`, `Refresh()` and `All()` functions
- A new `InsuranceCollection` class required for the `Insurance.All()` function
- Tests for all functionalities

### CODE REVIEW

File: EasyPost.Tests/InsuranceTest.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+namespace EasyPost.Tests
+{
+    [TestClass]
+    public class InsuranceTest
+    {
+        private Dictionary<string, object> _fromAddressData { get; set; }
+        private Insurance _insurance { get; set; }
+        private Dictionary<string, object> _parcelData { get; set; }
+        private Dictionary<string, object> _toAddressData { get; set; }
```
Comment (Justintime50) [inline EasyPost.Tests/InsuranceTest.cs:12]: Does the linter want these ordered alphabetically? If not, it'd make more sense to have from/to address next to each other, though I assume they're this way for the linter and that's fine.

File: EasyPost.Tests/InsuranceTest.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    "amount", (float)100
+                },
+                {
+                    "carrier", "USPS"
+                }
+            };
+        }
+
+        [TestMethod]
+        public void TestCreateAndRetrieve()
```
Comment (Justintime50) [inline EasyPost.Tests/InsuranceTest.cs:119]: Longterm, tests should only do one thing (test one thing) - We can leave this for now but when we circle back, there should be a standalone `create` test, a standalone `retrieve` test (which may still need to create an object, but it shouldn't assert against the created one, only the retrieved one), and then an `all` test aside.

File: EasyPost/Tracker.cs [hunk 3]
```diff
@@ -102,7 +102,12 @@ public static Tracker Retrieve(string id)
         public static bool CreateList(Dictionary<string, object> param)
         {
             Request request = new Request("trackers/create_list", RestSharp.Method.POST);
-            request.AddBody(new Dictionary<string, object>() { { "trackers", param } });
+            request.AddBody(new Dictionary<string, object>
+            {
+                {
+                    "trackers", param
```
Comment (Justintime50) [inline EasyPost/Tracker.cs:108]: `param` should be pluralized.

**Review by Justintime50 — APPROVED** [review]

File: EasyPost.Tests/InsuranceTest.cs (hunk 2 above)
Comment (nwithan8) [reply EasyPost.Tests/InsuranceTest.cs:119]: Almost every one of our test files has a `CreateAndRetrieve` method, so we'll need to rework that for the entire library and all tests.

File: EasyPost/Tracker.cs (hunk 3 above)
Comment (nwithan8) [reply EasyPost/Tracker.cs]: Resolved in [e01e5b4](https://github.com/EasyPost/easypost-csharp/pull/205/commits/e01e5b4a785997e20d948c2c0fccb4d2a0f0747b)

File: EasyPost.Tests/InsuranceTest.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+namespace EasyPost.Tests
+{
+    [TestClass]
+    public class InsuranceTest
+    {
+        private Dictionary<string, object> _fromAddressData { get; set; }
+        private Insurance _insurance { get; set; }
+        private Dictionary<string, object> _parcelData { get; set; }
+        private Dictionary<string, object> _toAddressData { get; set; }
+        private Dictionary<string, object> InsuranceData { get; set; }
```
Comment (jchen293) [inline EasyPost.Tests/InsuranceTest.cs:13]: **InsuranceData** is a private variable, should the variable start with an underscore and lowercase?

File: EasyPost.Tests/InsuranceTest.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                    "to_address", _toAddressData
+                },
+                {
+                    "from_address", _fromAddressData
+                },
+                {
+                    "tracking_code", shipment.tracking_code
+                },
+                {
+                    "amount", (float)100
```
Comment (jchen293) [inline EasyPost.Tests/InsuranceTest.cs:110]: Why are we converting **int** to **float** here? Can we just define it as float?

Comment (nwithan8) [reply EasyPost.Tests/InsuranceTest.cs:110]: Without the cast, it thinks it's an `int`. With decimals, it thinks it's a `double`

File: EasyPost.Tests/InsuranceTest.cs (hunk 4 above)
Comment (nwithan8) [reply EasyPost.Tests/InsuranceTest.cs:12]: Good point, resolved in [76da6a1](https://github.com/EasyPost/easypost-csharp/pull/205/commits/76da6a1c7800ac4df98f8c230458ee78540999bc)

File: EasyPost.Tests/InsuranceTest.cs (hunk 5 above)
Comment (nwithan8) [reply EasyPost.Tests/InsuranceTest.cs]: Resolved by using `decimal` type in [76da6a1](https://github.com/EasyPost/easypost-csharp/pull/205/commits/76da6a1c7800ac4df98f8c230458ee78540999bc)

File: EasyPost.Tests/InsuranceTest.cs (hunk 4 above)
Comment (jchen293) [reply EasyPost.Tests/InsuranceTest.cs:12]: Why are the private variables all start with uppercase now?

I am following the naming convention from the Microsoft doc: https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions

File: EasyPost/Insurance.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using RestSharp;
+
+namespace EasyPost
+{
+    public class Insurance : Resource
+    {
+        public decimal amount { get; set; }
```
Comment (Justintime50) [inline EasyPost/Insurance.cs:8]: This is a string (we use string fields almost exclusively with a few exceptions).

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: EasyPost.Tests/InsuranceTest.cs (hunk 4 above)
Comment (nwithan8) [reply EasyPost.Tests/InsuranceTest.cs:12]: These are properties (setters and getters), not fields.

File: EasyPost/Insurance.cs (hunk 6 above)
Comment (nwithan8) [reply EasyPost/Insurance.cs:8]: In the Java library, which I modeled this after, `amount` is a number, not a string: https://github.com/EasyPost/easypost-java/blob/master/src/main/java/com/easypost/model/Insurance.java#L22

File: EasyPost.Tests/InsuranceTest.cs (hunk 4 above)
Comment (jchen293) [reply EasyPost.Tests/InsuranceTest.cs:12]: thanks for the clarification.

File: EasyPost/Insurance.cs (hunk 6 above)
Comment (Justintime50) [reply EasyPost/Insurance.cs:8]: Unfortunately the API returns a string even if it was setup wrong in Java.

Comment (nwithan8) [reply EasyPost/Insurance.cs]: Resolved in [f696919](https://github.com/EasyPost/easypost-csharp/pull/205/commits/f696919c16d57f48faeb4f74bedd73f684112b54)

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**